engine/class_modules/sc_monk.cpp

```diff
@@ -4331,6 +4551,23 @@ struct monk_action_t : public Base
     }
   }
 
+  void trigger_empowered_tiger_lightning( action_state_t* s )
+  {
+    if ( p()->spec.invoke_xuen_2->ok() )
+    {
+      // Make sure Xuen is up and the action is not the Empowered Tiger Lightning itself
+      // Touch of Karma (id = 124280) does not contribute to Empowered Tiger Lightning
+      if ( p()->buff.invoke_xuen->up() && s->result_total > 0 
```

Comment:
Instead of checking for ids, would it be possible to add a flag
bool triggers_invoke_xuen_2; to monk_action_t, which you could then disable on specific actions in the constructor by adding triggers_invoke_xuen_2 = false;